src/pulp/solvers.py

```diff
@@ -1275,75 +1278,94 @@ def actualSolve(self, lp):
+            xpress.stdin.write("HEURSTRATEGY=%d\n" % self.heurStra)
         if self.coverCuts:
-            xpress.write("COVERCUTS=%d\n" % self.coverCuts)
+            xpress.stdin.write("COVERCUTS=%d\n" % self.coverCuts)
         if self.preSolve:
-            xpress.write("PRESOLVE=%d\n" % self.preSolve)
+            xpress.stdin.write("PRESOLVE=%d\n" % self.preSolve)
         for option in self.options:
-            xpress.write(option+"\n")
+            xpress.stdin.write(option+"\n")
         if lp.sense == LpMaximize:
-            xpress.write("MAXIM\n")
+            xpress.stdin.write("MAXIM\n")
         else:
-            xpress.write("MINIM\n")
+            xpress.stdin.write("MINIM\n")
         if lp.isMIP() and self.mip:
-            xpress.write("GLOBAL\n")
-        xpress.write("WRITEPRTSOL "+tmpSol+"\n")
-        xpress.write("QUIT\n")
-        if xpress.close() != None:
+            xpress.stdin.write("GLOBAL\n")
+        if self.fullPrecision:
+            xpress.stdin.write("WRITESOL "+tmpSol+ " -p\n")
+        else:
+            xpress.stdin.write("WRITEPRTSOL "+tmpSol+"\n")
+        xpress.stdin.write("QUIT\n")
+        xpress.stdin.close()
+        if xpress.wait() != 0:
             raise PulpSolverError("PuLP: Error while executing "+self.path)
         status, values = self.readsol(tmpSol)
         if not self.keepFiles:
             try: os.remove(tmpLp)
             except: pass
             try: os.remove(tmpSol)
             except: pass
+        lp.status = status
         lp.assignVarsVals(values)
-        # TODO: the following could be put inside the assignStatus function
         if abs(lp.infeasibilityGap(self.mip)) > 1e-5: # Arbitrary
-            status = LpStatusInfeasible
-        lp.assignStatus(status)
-        return status
+            lp.status = LpStatusInfeasible
+        return lp.status
 
     def readsol(self,filename):
         """Read an XPRESS solution file"""
-        with open(filename) as f:
-            for i in range(6): f.readline()
-            l = f.readline().split()
-
-            rows = int(l[2])
-            cols = int(l[5])
-            for i in range(3): f.readline()
-            statusString = f.readline().split()[0]
-            # TODO: check status for Integer Feasible
-            xpressStatus = {
-                "Optimal":LpStatusOptimal,
-                }
-            if statusString not in xpressStatus:
-                raise PulpSolverError("Unknown status returned by XPRESS: "+statusString)
-            status = xpressStatus[statusString]
-            values = {}
-            while 1:
-                l = f.readline()
-                if l == "": break
-                line = l.split()
-                if len(line) and line[0] == 'C':
-                    name = line[2]
-                    value = float(line[4])
-                    values[name] = value
-        return status, values
+        if self.fullPrecision:
```

Comment:
I think this is better if you make a new function (`readsol_fullprecision`) and leave `readsol` as it is.